### PR TITLE
Add C++ torch::nn::functional::pdist

### PR DESCRIPTION
**Pull request by jon-tow**
Adds `torch::nn::functional::pdist` module support for the C++ API.

Issue: #25883, #27082 

Reviewer: @yf225

### CODE REVIEW

File: torch/csrc/api/include/torch/nn/functional/distance.h [hunk 1]
```diff
@@ -31,6 +31,14 @@ inline Tensor pairwise_distance(
       options.keepdim());
 }
 
+// ============================================================================
+
+/// Computes the p-norm distance between every pair of row vectors in the input.
+/// This function will be faster if the rows are contiguous.
+inline Tensor pdist(const Tensor& input, int64_t p = 2) {
```
Comment (yf225) [inline torch/csrc/api/include/torch/nn/functional/distance.h]: @jon-tow My sincere apologies - I made a mistake and just realized that `p` should be `double` type (https://en.wikipedia.org/wiki/Norm_(mathematics)#p-norm). Sorry for the inconvenience :(

Comment (jon-tow) [reply torch/csrc/api/include/torch/nn/functional/distance.h]: @yf225 Ah wow, good catch. I was also certain that `p` was integer rather than real.

**Review by yf225 — APPROVED** [review]
@jon-tow Thanks for the great work! I will merge it first thing tomorrow morning :)

**Comment by jon-tow** [comment]
@yf225 You're welcome!

**Pull request closed without merging**